client/src/pages/ChatPage.js

```diff
@@ -0,0 +1,315 @@
+        <SideBar />
+      </div>
+      <div className="p-5 !pl-[300px]">
+        <div className="flex justify-center">
+          <Chip
+            value={chat.name}
+            variant="outlined"
+            className="text-center text-indigo-300 w-3/5"
+          />
+        </div>
+
+        <div className="absolute top-0 right-0 m-5">
+          <IconButton
+            type="submit"
+            onClick={changeNameHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <PencilSquareIcon className="h-5 w-5 " />
+          </IconButton>
+
+          <IconButton
+            type="submit"
+            onClick={addUserHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <UserPlusIcon className="h-5 w-5" />
+          </IconButton>
+
+          <IconButton
+            type="submit"
+            onClick={exitUserHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <ArrowRightOnRectangleIcon className="h-5 w-5" />
+          </IconButton>
+        </div>
+
+        <div className="flex justify-center">
+          <div className="mb-5 absolute bottom-0 w-3/5">
+            <Card className="flex mb-5">
+              <ul className="p-5 max-h-[30rem] overflow-y-auto scrollbar-width:none">
+                {chat.history
+                  ? chat.history.map((message, index) => (
+                      <li key={index}>
+                        <span className="font-bold">{message.username}: </span>
+                        {message.message}
+                      </li>
+                    ))
+                  : null}
+              </ul>
+            </Card>
+            <Card className="flex">
+              <form onSubmit={handleSubmit} className="flex p-5">
+                <Textarea
+                  label="Message"
+                  value={message}
+                  onChange={(event) => setMessage(event.target.value)}
+                />
+
+                <div className="flex items-center">
+                  <IconButton type="submit" className="flex bg-indigo-300 ml-5">
+                    <PaperAirplaneIcon className="h-3 w-3" />
+                  </IconButton>
+                </div>
+              </form>
+            </Card>
+          </div>
+        </div>
+      </div>
+
+      <Dialog
+        size="sm"
+        open={openUser}
+        handler={addUserHandler}
+        className="bg-transparent shadow-none"
+      >
+        <Card className="mx-auto w-full max-w-[24rem]">
+          <CardHeader
+            variant="gradient"
+            className="mb-4 grid h-28 place-items-center bg-indigo-300"
+          >
+            <Typography variant="h3" color="white">
+              Invite Friend
+            </Typography>
+          </CardHeader>
+          <CardBody className="flex flex-col gap-4">
+            <Input label="Username" size="lg" onChange={onInviteUser} />
```

Comment:
In the future there should probably be some user feedback on whether or not the invite was successful. You can also currently invite yourself (but this does nothing). Inviting a non-existent user displays a 400 error in console which could be used to discover usernames within the app:

![image](https://github.com/awctw/study-dash/assets/56493743/1b37284f-dcab-42cb-8bd1-f56ea1d5d849)